controllers/watcherapi_controller.go

```diff
@@ -272,6 +306,58 @@ func (r *WatcherAPIReconciler) generateServiceConfigs(
 	return GenerateConfigsGeneric(ctx, helper, instance, envVars, templateParameters, customData, labels, false)
 }
 
+func (r *WatcherAPIReconciler) createDeployment(
+	ctx context.Context,
+	helper *helper.Helper,
+	instance *watcherv1beta1.WatcherAPI,
+	configHash string,
+) (ctrl.Result, error) {
+	Log := r.GetLogger(ctx)
+	Log.Info(fmt.Sprintf("Defining WatcherAPI deployment '%s'", instance.Name))
+
+	labels := labels.GetLabels(instance, labels.GetGroupLabel(watcher.ServiceName), map[string]string{})
+
+	// define a new Deployment object
+	deploymentDef, err := watcherapi.Deployment(instance, configHash, labels)
+	if err != nil {
+		instance.Status.Conditions.Set(condition.FalseCondition(
+			condition.DeploymentReadyCondition,
+			condition.ErrorReason,
+			condition.SeverityWarning,
+			condition.DeploymentReadyErrorMessage,
+			err.Error()))
+		return ctrl.Result{}, err
+	}
+	Log.Info(fmt.Sprintf("Getting deployment '%s'", instance.Name))
+	deploymentObject := deployment.NewDeployment(deploymentDef, time.Duration(5)*time.Second)
+	Log.Info(fmt.Sprintf("Got deployment '%s'", instance.Name))
+	ctrlResult, errorResult := deploymentObject.CreateOrPatch(ctx, helper)
+	if errorResult != nil {
+		instance.Status.Conditions.Set(condition.FalseCondition(
+			condition.DeploymentReadyCondition,
+			condition.ErrorReason,
+			condition.SeverityWarning,
+			condition.DeploymentReadyErrorMessage,
+			errorResult.Error()))
+		return ctrlResult, errorResult
+	} else if (ctrlResult != ctrl.Result{}) {
+		instance.Status.Conditions.Set(condition.FalseCondition(
+			condition.DeploymentReadyCondition,
+			condition.RequestedReason,
+			condition.SeverityInfo,
+			condition.DeploymentReadyRunningMessage))
+		return ctrlResult, nil
+	}
+
+	instance.Status.ReadyCount = deploymentObject.GetDeployment().Status.ReadyReplicas
+
+	if instance.Status.ReadyCount > 0 {
+		instance.Status.Conditions.MarkTrue(condition.DeploymentReadyCondition, condition.DeploymentReadyMessage)
+	}
```

Comment:
Adding an `else` here I think would make sense to set the condition to "not ready", i.e. [1].

[1] https://github.com/openstack-k8s-operators/keystone-operator/blob/8e243bd3659690a7f3a11bd48c7be29552dac7fe/controllers/keystoneapi_controller.go#L1071-L1079

Reply:
good idea, thanks, added it